code/modules/ninja/suit/suit.dm

```diff
@@ -164,45 +224,15 @@ Contents:
 		n_gloves.candrain = FALSE
 		n_gloves.draining = FALSE
 
+		REMOVE_TRAIT(ninja, TRAIT_NOGUNS, NINJA_SUIT_TRAIT)
 
-/obj/item/clothing/suit/space/space_ninja/examine(mob/user)
-	. = ..()
-	if(s_initialized)
-		if(user == affecting)
-			. += "All systems operational. Current energy capacity: <B>[DisplayEnergy(cell.charge)]</B>.\n"+\
-			"The CLOAK-tech device is <B>[stealth?"active":"inactive"]</B>.\n"+\
-			"There are <B>[s_bombs]</B> smoke bomb\s remaining.\n"+\
-			"There are <B>[a_boost]</B> adrenaline booster\s remaining."
-
-/obj/item/clothing/suit/space/space_ninja/ui_action_click(mob/user, action)
-	if(istype(action, /datum/action/item_action/initialize_ninja_suit))
-		toggle_on_off()
-		return TRUE
-	if(!s_initialized)
-		to_chat(user, "<span class='warning'><b>ERROR</b>: suit offline. Please activate suit.</span>")
-		return FALSE
-	if(istype(action, /datum/action/item_action/ninjasmoke))
-		ninjasmoke()
-		return TRUE
-	if(istype(action, /datum/action/item_action/ninjaboost))
-		ninjaboost()
-		return TRUE
-	if(istype(action, /datum/action/item_action/ninjapulse))
-		ninjapulse()
-		return TRUE
-	if(istype(action, /datum/action/item_action/ninjastar))
-		ninjastar()
-		return TRUE
-	if(istype(action, /datum/action/item_action/ninjanet))
-		ninjanet()
-		return TRUE
-	if(istype(action, /datum/action/item_action/ninja_sword_recall))
-		ninja_sword_recall()
-		return TRUE
-	if(istype(action, /datum/action/item_action/ninja_stealth))
-		stealth()
-		return TRUE
-	if(istype(action, /datum/action/item_action/toggle_glove))
-		n_gloves.toggledrain()
-		return TRUE
-	return FALSE
+/**
+  * Proc used to delete all the attachments and itself.
+  *
+  * Can be called to entire rid of the suit pieces and the suit itself.
+  */
+/obj/item/clothing/suit/space/space_ninja/proc/terminate()
+	qdel(n_hood)
+	qdel(n_gloves)
+	qdel(n_shoes)
```

Comment:
don't keep references to deleted thing, else you'll force a hard delete
null the variables
you have `QDEL_NULL` for that